op-service/sources/supervisor_client.go

```diff
@@ -34,40 +35,57 @@ type SupervisorQueryAPI interface {
 }
 
 type SupervisorClient struct {
-	client client.RPC
+	client  client.RPC
+	metrics opmetrics.RPCClientMetricer
 }
 
 // This type-check keeps the Server API and Client API in sync.
 var _ SupervisorQueryAPI = (*SupervisorClient)(nil)
 var _ SupervisorAdminAPI = (*SupervisorClient)(nil)
 
-func NewSupervisorClient(client client.RPC) *SupervisorClient {
+func NewSupervisorClient(client client.RPC, metrics opmetrics.RPCClientMetricer) *SupervisorClient {
+	if metrics == nil {
+		metrics = &opmetrics.NoopRPCMetrics{}
+	}
 	return &SupervisorClient{
-		client: client,
+		client:  client,
+		metrics: metrics,
 	}
 }
 
 func (cl *SupervisorClient) Stop(ctx context.Context) error {
+	done := cl.metrics.RecordRPCClientRequest("admin_stop")
+	var err error
+	defer func() { done(err) }()
```

Comment:
nit: these "done" sections could be condensed too:

```
var err error
defer cl.metrics.RecordRPCClientRequest("admin_stop")(err)
```

Reply:
@axelKingsley I think this doesn't work because the value of `err` that is passed to the defer'd function is bound at the time we call defer, before we set `err`.